app/src/main/java/com/sdpteam/connectout/event/viewer/EventActivity.java

```diff
@@ -131,6 +141,14 @@ private void updateEventView(Event event, TextView title, TextView description,
         chatBtn.setVisibility(event.hasJoined(currentUserId) || event.isInterested(currentUserId) ? VISIBLE : GONE);
         chatBtn.setOnClickListener(v -> openChat(event.getId()));
 
+        shareQrCodeBtn.setOnClickListener(v -> {
+            String qrCodeData = "event/" + event.getId();
+            Intent intent = new Intent(EventActivity.this, QRcodeModalActivity.class);
+            intent.putExtra("title", "Event QR code");
+            intent.putExtra("qrCodeData", qrCodeData);
+            qrCodeLauncher.launch(intent);
```

Comment:
The whole method is already quite big, I would suggest to extract a separate private method for that